src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -144,5 +164,47 @@ public bool Equals(ReadOnlySpan<T> other)
             return (_length == other._length) &&
                 (_length == 0 || JitHelpers.ByRefEquals(ref JitHelpers.GetByRef<T>(ref _rawPointer), ref JitHelpers.GetByRef<T>(ref other._rawPointer)));
         }
+
+        /// <summary>
+        /// Copies the contents of this span into destination span. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The span to copy items into.</param>
+        public bool TryCopyTo(Span<T> destination)
+        {
+            if (Length > destination.Length)
+                return false;
+
+            SpanHelper.CopyTo<T>(ref destination._rawPointer, ref _rawPointer, Length);
+            return true;
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination array. The destination
+        /// must be at least as big as the source, and may be bigger.
+        /// </summary>
+        /// <param name="destination">The array to copy items into.</param>
+        public bool TryCopyTo(T[] destination)
+        {
+            return TryCopyTo(new Span<T>(destination));
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into destination memory. 
+        /// Only Value Types that contain no pointers are supported.
+        /// </summary>
+        /// <param name="destination">An unmanaged pointer to memory.</param>
+        /// <param name="elementsCount">The number of T elements the memory contains.</param>
+        [CLSCompliant(false)]
+        public unsafe bool TryCopyTo(void* destination, int elementsCount)
```

Comment:
To me it's about ensuring that the caller does not cause buffer overruns because they misjudged the size of the buffer needed to fit the elements. It's much less likely that they don't know how many bytes they have in the destination.

@jkotas, any opinions on this?